src/main/java/org/gridsuite/explore/server/services/ExploreService.java

```diff
@@ -143,42 +152,36 @@ public ExploreService(
 
     public Mono<Void> deleteElement(UUID id, String userId) {
         return directoryService.getElementInfos(id).flatMap(elementAttributes -> {
-            if (elementAttributes.getType().equals(STUDY)) {
-                studyService.deleteFromStudyServer(elementAttributes.getElementUuid(), userId).subscribe();
-            } else if (elementAttributes.getType().equals(CONTINGENCY_LIST)) {
-                contingencyListService.deleteContingencyList(elementAttributes.getElementUuid()).subscribe();
-            } else if (elementAttributes.getType().equals(FILTER)) {
-                filterService.deleteFilter(elementAttributes.getElementUuid()).subscribe();
+            IDirectoryElementsService service = genericServices.get(elementAttributes.getType());
+            if (service == null) {
+                return Mono.error(new ExploreException(UNKNOWN_ELEMENT_TYPE, "Unknown element type :" + elementAttributes.getType()));
             }
-            return directoryService.deleteElement(id, userId);
+            return service.delete(id, userId).doOnSuccess(e -> directoryService.deleteElement(id, userId).subscribe());
         });
     }
 
-    public Mono<List<ElementAttributes>> getElementsMetadata(List<UUID> ids) {
+    public Flux<ElementAttributes> getElementsMetadata(List<UUID> ids) {
         Mono<Map<UUID, ElementAttributes>> elementsAttributesMono = directoryService.getElementsAttribute(ids).collect(Collectors.toMap(ElementAttributes::getElementUuid, Function.identity()));
+        return elementsAttributesMono.flatMapMany(elementsAttributes -> {
```

Comment:
I think it is too complex to mix stream et flux
why we do not use `GroupedFlux` and use only flux ?
`        Flux<GroupedFlux<String, ElementAttributes>> elementsAttributesFlux = directoryService.getElementsAttribute(ids).groupBy(e -> e.getType());
`

Reply:
done